### DISCUSSION TITLE
Multiple state changes are not reflected in the application on a single button click

### DISCUSSION THREAD

**Discussion by abhinavsns** [Kitware/trame#407] [Trame/ParaView] [answered]
Hi,

I have vuetify3 button that changes a state variable multiple times.

```
def on_button_click():
    state.message="On"
    time.sleep(5)
    state.message="Off"

vuetify.VBtn(
    "Click me",
    click=on_button_click
)
```

The state.message is observed by another Veutify field but I only see off being reflected when the button is clicked. This issue is not resolved when I use another thread to change the state to off. Even using state.flush() after "On". I still do not see "On" on the application.  

Is there a simple way of reflecting multiple state changes with a single button click using trame-paraview?

Thank you.

**Comment by jourdain** [comment, marked as answer]
You have to remember that you are dealing with a client/server architecture, therefore, you need to provide freedom to the network layer to do its thing. 

You can look at that previously discussed topic with a solution [here](https://github.com/Kitware/trame/discussions/274).

**Reply by abhinavsns** [reply]
Thanks! I will keep that in mind. Still learning this excellent framework.
